src/solver/variable/state.h

```diff
@@ -37,23 +37,40 @@
 
 namespace Antares::Solver::Variable
 {
+class ThermalState
+{
+public:
+    explicit ThermalState(const Data::AreaList& areas);
+
+    class StateForAnArea
+    {
+    public:
+        void initializeFromArea(const Data::Area& area);
+        //! Thermal production for thermal clusters for the current hour in the year
+        std::vector<double> thermalClustersProductions;
+
+        //! The operating cost for all clusters at the current hour (production level*production
+        //! cost + NP Cost)
+        std::vector<double> thermalClustersOperatingCost;
+
+        //! Number of units turned ON by cluster for the current hour in the year with the ucMILP
+        //! (accurate) unit commitment mode
+        std::vector<uint> numberOfUnitsONbyCluster;
+
+        //! Minimum power of all clusters for the current hour in the year
+        std::vector<double> PMinOfClusters;
+    };
+
+    StateForAnArea& operator[](size_t index);
```

Comment:
```C++
StateForAnArea& operator[](size_t index);
```
I guess the argument is an index for an area, right ?
Could we rename this argument **area_index** ?